subsys/bluetooth/audio/unicast_client.c

```diff
@@ -2592,7 +2595,12 @@ int bt_audio_discover(struct bt_conn *conn,
 	params->num_caps = 0u;
 	params->num_eps = 0u;
 
-	return bt_gatt_read(conn, &params->read);
+	err = bt_gatt_read(conn, &params->read);
+	if (err == 0) {
+		unicast_client_pac_reset(conn, params->dir);
```

Comment:
Wouldn't be easier to just reset PAC's unconditionally?

Reply:
Possibly, but if discovery start fails, there's no reason to do it. There is a small performance gain on doing this conditionally, but I am OK with doing it unconditionally.

Reply:
I would like to handle the discovery as a single procedure (even though it is an (or many) async operations). 

I'll move the reset to before the `bt_gatt_read`